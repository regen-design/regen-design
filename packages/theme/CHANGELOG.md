# @regen-design/theme

## 1.0.0

### Major Changes

- publish
