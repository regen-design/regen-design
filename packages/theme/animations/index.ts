import { keyframes } from 'styled-components'

export const ButtonWaveSpreadAnimationKeyframes = (color: string) => keyframes`
    0% {
        opacity: 0.6;
        box-shadow: 0 0 1px 0 ${color};
    }
    100% {
        opacity: 0;
        box-shadow: 0 0 1px 5px ${color};
    }
`
