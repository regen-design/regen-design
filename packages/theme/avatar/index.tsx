import { NAME_SPACE } from '@regen-design/constant'
import { AvatarProps } from '@regen-design/types'
import styled from 'styled-components'
import { convertTheme } from '../tools'
const prefix = `${NAME_SPACE}-avatar`
export const StyledAvatarPrefixClass = prefix
export const StyledAvatar = styled.div<AvatarProps>`
  display: inline-flex;
  position: relative;
  overflow: hidden;
  text-align: center;
  border: ${props => {
    const _theme = convertTheme(props.theme)
    return `1px solid ${_theme.colors[props.borderColor || 'transparent']}`
  }};
  width: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.components.avatar.size[props.size || 'default']
  }};
  height: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.components.avatar.size[props.size || 'default']
  }};
  color: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.colors[props.color || 'white']
  }};
  font-size: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.components.avatar.fontSize
  }};
  border-radius: ${props => {
    const _theme = convertTheme(props.theme)
    if (props.rounded) return '50%'
    return _theme.borderRadius
  }};
  background-color: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.colors[props.backgroundColor || 'placeholder']
  }};
  img {
    width: 100%;
    height: 100%;
  }
  .${prefix}__text {
    position: absolute;
    line-height: 1;
    white-space: nowrap;
    display: inline-block;
    left: 50%;
    top: 50%;
  }
`
