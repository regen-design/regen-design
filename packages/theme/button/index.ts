import styled from 'styled-components'
import { convertTheme } from '../tools'
import { ButtonProps } from '@regen-design/types'
import { NAME_SPACE } from '@regen-design/constant'
import { lighten } from 'polished'
const prefix = `${NAME_SPACE}-button`

export const StyledButtonPrefixClass = prefix

export const StyledButton = styled.button<ButtonProps>`
  background-color: ${props => {
    const _theme = convertTheme(props.theme)
    if (props.text) {
      return 'transparent'
    }
    if (props.type === 'default') {
      return _theme.theme === 'light' ? _theme.colors.light : _theme.colors.dark
    }
    if (props.dashed) {
      return 'transparent'
    }
    return _theme.colors[props.type]
  }};
  border-radius: ${props => {
    return convertTheme(props.theme).borderRadius + 'px'
  }};
  color: ${props => {
    const _theme = convertTheme(props.theme)
    if (props.text) {
      return _theme.colors[props.type]
    }
    if (props.type === 'default') {
      return _theme.theme === 'light' ? _theme.colors.dark : _theme.colors.light
    }
    if (props.dashed) {
      return `${_theme.colors[props.type]}`
    }
    return _theme.theme === 'light' ? _theme.colors.light : _theme.colors.dark
  }};
  &:hover {
    color: ${props => {
      const _theme = convertTheme(props.theme)
      if (props.type === 'default') {
        return lighten(0.025, _theme.colors.primary)
      }
      if (props.text) {
        return lighten(0.05, _theme.colors?.[props.type] || '#000000')
      }
    }};
  }
  &:active {
    color: ${props => {
      const _theme = convertTheme(props.theme)
      if (props.type === 'default') {
        return lighten(0.15, _theme.colors.primary)
      }
      if (props.text) {
        return lighten(0.25, _theme.colors?.[props.type] || '#000000')
      }
      if (props.dashed) {
        return `${_theme.colors[props.type]}`
      }
      return _theme.theme === 'light' ? _theme.colors.light : _theme.colors.dark
    }};
  }
  transition: ${props => {
    const duration = `${convertTheme(props.theme).animations.duration}ms`
    return `all ${props.text ? '50ms' : duration} ease-in`
  }};
  opacity: ${props => {
    if (props.disabled) {
      return 0.5
    }
  }};
  display: inline-flex;
  flex-wrap: nowrap;
  white-space: nowrap;
  outline: none;
  position: relative;
  z-index: auto;
  border: ${props => {
    const _theme = convertTheme(props.theme)
    if (props.text) {
      return 'transparent'
    }
    if (props.type === 'default') {
      const color = _theme.theme === 'light' ? _theme.borderColor : _theme.borderDarkColor
      return `1px ${props.dashed ? 'dashed' : 'solid'} ${color}`
    }
    if (props.dashed) {
      return `1px dashed ${_theme.colors[props.type]}`
    }
    return '1px solid transparent'
  }};
  padding: ${props => {
    if (props.text) {
      return '0 0'
    }
    switch (props.size) {
      case 'tiny':
        return '0 8px'
      case 'small':
        return '0 12px'
      case 'large':
        return '0 16px'
      default:
        return '0 12px'
    }
  }};
  height: ${props => {
    if (props.text) {
      return 'initial'
    }
    switch (props.size) {
      case 'tiny':
        return '24px'
      case 'small':
        return '32px'
      case 'large':
        return '40px'
      default:
        return '36px'
    }
  }};
  line-height: 1;
  font-family: inherit;
  font-size: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.fontSizes[props.size]
  }};
  width: initial;
  align-items: center;
  justify-content: center;
  user-select: none;
  -webkit-user-select: none;
  text-align: center;
  cursor: ${props => {
    if (props.disabled) {
      return 'not-allowed'
    }
    return 'pointer'
  }};
  text-decoration: none;
  .${prefix}__content {
    display: flex;
    align-items: center;
    flex-wrap: nowrap;
    min-width: 0;
  }
  .${prefix}__icon {
    margin-right: 8px;
  }
`
