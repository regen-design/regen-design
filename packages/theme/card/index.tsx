import { NAME_SPACE } from '@regen-design/constant'
import { CardProps } from '@regen-design/types'
import styled from 'styled-components'
import { convertTheme } from '../tools'
const prefix = `${NAME_SPACE}-card`
export const StyledCardPrefixClass = prefix
export const StyledCard = styled.div<CardProps>`
  border: ${props => {
    const _theme = convertTheme(props.theme)
    return `1px solid ${_theme.theme === 'light' ? _theme.borderColor : _theme.borderDarkColor}`
  }};
  border-radius: ${props => convertTheme(props.theme).borderRadius};
  border-bottom: 0;
  width: 100%;
  box-sizing: border-box;
  position: relative;
  > .${prefix}-header {
    padding: ${props => {
      const _theme = convertTheme(props.theme)
      return `${_theme.paddingSizes[props.size]}`
    }};
    display: flex;
    justify-content: space-between;
    align-items: center;
    > .${prefix}-title {
      font-size: ${props => {
        const _theme = convertTheme(props.theme)
        return `${_theme.fontSizes[props.size]}`
      }};
      color: ${props => props.theme.textColor};
    }
  }

  > .${prefix}-body {
    > .${prefix}-description {
      margin-bottom: ${props => {
        const _theme = convertTheme(props.theme)
        return `${_theme.paddingSizes[props.size]}`
      }};
    }
    padding: ${props => {
      const _theme = convertTheme(props.theme)
      const size = _theme.paddingSizes[props.size]
      return `0 ${size} ${size}`
    }};
    border-bottom: ${props => {
      const _theme = convertTheme(props.theme)
      return `1px solid ${_theme.theme === 'light' ? _theme.borderColor : _theme.borderDarkColor}`
    }};
  }
  > .${prefix}-footer {
    position: relative;
    border-bottom: ${props => {
      const _theme = convertTheme(props.theme)
      const color = _theme.theme === 'dark' ? _theme.borderDarkColor : _theme.borderColor
      return `1px solid ${color}`
    }};
    padding: ${props => {
      const _theme = convertTheme(props.theme)
      const size = _theme.paddingSizes[props.size]
      return `${size}`
    }};
  }
`
