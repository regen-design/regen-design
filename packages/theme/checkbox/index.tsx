import { NAME_SPACE } from '@regen-design/constant'
import { CheckboxProps } from '@regen-design/types'
import styled from 'styled-components'
import { convertTheme } from '../tools'
import { lighten, rgba } from 'polished'

const prefix = `${NAME_SPACE}-checkbox`
export const StyledCheckboxPrefixClass = prefix
export const StyledCheckbox = styled.div<CheckboxProps>`
  display: inline-flex;
  flex-wrap: nowrap;
  word-break: break-word;
  align-items: flex-start;
  line-height: 1.5;
  font-size: ${props => {
    const _theme = convertTheme(props.theme)
    return _theme.fontSizes[props.size]
  }};
  cursor: ${props => {
    if (props.disabled) {
      return 'not-allowed'
    }
    return 'pointer'
  }};
  margin: 0;
  padding: 0;
  &:not(.${prefix}--disabled):focus {
    .${prefix}-wrapper {
      .${prefix}-box__border {
        border: ${props => {
          const _theme = convertTheme(props.theme)
          return `1px solid ${_theme.colors['primary']}`
        }};
      }
      .${prefix}-box {
        box-shadow: ${props => {
          const _theme = convertTheme(props.theme)
          const color = _theme.colors.primary
          return `0 0 0 ${_theme.waveBlurRadius / 2}px ${rgba(color, 0.15)}`
        }};
      }
    }
  }
  .${prefix}-label {
    user-select: none;
    -webkit-user-select: none;
    padding: 0 4px;
    line-height: 1.5;
    color: ${props => {
      const _theme = convertTheme(props.theme)
      if (props.disabled) {
        return _theme.colors.placeholder
      }
      return 'inherit'
    }};
  }
  .${prefix}-wrapper {
    .${prefix}-box {
      background-color: ${props => {
        const _theme = convertTheme(props.theme)
        if (props.disabled) {
          if (_theme.theme === 'dark') {
            return rgba('#ffffff', 0.1)
          }
          return _theme.colors.light
        }
        if (props.indeterminate || props.checked) {
          return _theme.colors.primary
        }
        return 'inherit'
      }};
      transition:
        background-color 0.3s ease,
        box-shadow 0.3s ease;
      border-radius: inherit;
      width: ${props => {
        const _theme = convertTheme(props.theme)
        return _theme.fontSizes[props.size]
      }};
      height: ${props => {
        const _theme = convertTheme(props.theme)
        return _theme.fontSizes[props.size]
      }};
      position: absolute;
      left: 0;
      top: 50%;
      transform: translateY(-50%);
    }
    .${prefix}-icon {
      color: ${props => {
        const _theme = convertTheme(props.theme)
        if (props.disabled) {
          if (_theme.theme === 'dark') {
            return lighten(0.25, _theme.colors.dark)
          }
          return _theme.colors.placeholder
        }
        return _theme.colors.light
      }};
      transform-origin: center center;
      display: flex;
      align-items: center;
      justify-content: center;
      position: absolute;
      width: 100%;
      height: 100%;
      padding: 2px;
      font-weight: bold;
      font-size: ${props => {
        const _theme = convertTheme(props.theme)
        return `calc(${_theme.fontSizes[props.size]} )`
      }};
    }
    .${prefix}-box__border {
      width: 100%;
      height: 100%;
      border-radius: inherit;
      border: 1px solid
        ${props => {
          const _theme = convertTheme(props.theme)
          if (_theme.theme === 'dark') {
            return _theme.borderDarkColor
          }
          if (props.disabled) {
            return _theme.colors.placeholder
          }
          if (props.indeterminate || props.checked) {
            return _theme.colors.primary
          }
          return _theme.borderColor
        }};
      transition: border-color 0.3s ease;
    }
    box-sizing: border-box;
    position: relative;
    flex-shrink: 0;
    flex-grow: 0;
    user-select: none;
    width: ${props => {
      const _theme = convertTheme(props.theme)
      return _theme.fontSizes[props.size]
    }};
    -webkit-user-select: none;
    white-space: nowrap;
    display: inline-block;
    line-height: 1.5;
    border-radius: ${props => {
      const _theme = convertTheme(props.theme)
      return _theme.borderRadius
    }};
  }
`
