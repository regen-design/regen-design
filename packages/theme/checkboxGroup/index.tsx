import { NAME_SPACE } from '@regen-design/constant'
import { CheckboxGroupProps } from '@regen-design/types'
import styled from 'styled-components'
const prefix = `${NAME_SPACE}-checkboxGroup`
export const StyledCheckboxGroupPrefixClass = prefix
export const StyledCheckboxGroup = styled.div<CheckboxGroupProps>``
