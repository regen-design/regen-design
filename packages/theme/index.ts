import { ThemeConfig } from '@regen-design/types'
import { useTheme as useThemeFn } from 'styled-components'
export const useTheme = useThemeFn as () => ThemeConfig
export * from './animations'
export * from './button'
export * from './card'
export * from './checkbox'
export * from './checkboxGroup'
export * from './codeBlock'
export * from './grid'
export * from './gridItem'
export * from './icon'
export * from './input'
export * from './radio'
export * from './radioGroup'
export * from './space'
export * from './variable'
export * from './wave'
