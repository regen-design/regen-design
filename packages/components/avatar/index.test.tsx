import { describe, expect, test } from 'vitest'
import '@testing-library/jest-dom/vitest'
import { render, screen } from '@testing-library/react'
import { Avatar } from '.'
import { defaultTheme } from '@regen-design/theme'

describe('Avatar', () => {
  test('should render correctly', () => {
    const element = render(<Avatar />)
    expect(element).toMatchSnapshot()
    expect(screen.getByTestId('avatar')).toBeInTheDocument()
  })

  test('should render children', () => {
    const element = render(<Avatar>Avatar</Avatar>)
    expect(element).toMatchSnapshot()
    expect(screen.getByText('Avatar')).toBeInTheDocument()
  })

  test('should render src', () => {
    const element = render(
      <Avatar src="https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50" />
    )
    expect(element).toMatchSnapshot()
    expect(screen.getByTestId('avatar').querySelector('img')).toHaveAttribute(
      'src',
      'https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50'
    )
  })

  test('should render size', () => {
    const element = render(<Avatar size={'default'} />)
    expect(element).toMatchSnapshot()
    expect(screen.getByTestId('avatar')).toHaveStyle(
      `width: ${defaultTheme.components.avatar.size['default']}`
    )
    expect(screen.getByTestId('avatar')).toHaveStyle(
      `height: ${defaultTheme.components.avatar.size['default']}`
    )
  })

  test('should render rounded', () => {
    const element = render(<Avatar rounded />)
    expect(element).toMatchSnapshot()
    expect(screen.getByTestId('avatar')).toHaveStyle('border-radius: 50%')
  })

  test('should render style', () => {
    const element = render(<Avatar style={{ color: 'red' }} />)
    expect(element).toMatchSnapshot()
    expect(screen.getByTestId('avatar')).toHaveStyle('color: red')
  })

  test('should render className', () => {
    const element = render(<Avatar className="test" />)
    expect(element).toMatchSnapshot()
    expect(screen.getByTestId('avatar')).toHaveClass('test')
  })
})
