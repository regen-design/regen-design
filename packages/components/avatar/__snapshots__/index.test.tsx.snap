// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar > should render children 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-beySPh fPcNGa regen-avatar"
        color="white"
        data-testid="avatar"
        role="avatar"
      >
        <span
          class="regen-avatar__text"
          style="transform: translateX(-50%) translateY(-50%) scale(0);"
        >
          Avatar
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-beySPh fPcNGa regen-avatar"
      color="white"
      data-testid="avatar"
      role="avatar"
    >
      <span
        class="regen-avatar__text"
        style="transform: translateX(-50%) translateY(-50%) scale(0);"
      >
        Avatar
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar > should render className 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-beySPh fPcNGa regen-avatar test"
        color="white"
        data-testid="avatar"
        role="avatar"
      >
        <span
          class="regen-avatar__text"
          style="transform: translateX(-50%) translateY(-50%) scale(1);"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-beySPh fPcNGa regen-avatar test"
      color="white"
      data-testid="avatar"
      role="avatar"
    >
      <span
        class="regen-avatar__text"
        style="transform: translateX(-50%) translateY(-50%) scale(1);"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-beySPh fPcNGa regen-avatar"
        color="white"
        data-testid="avatar"
        role="avatar"
      >
        <span
          class="regen-avatar__text"
          style="transform: translateX(-50%) translateY(-50%) scale(1);"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-beySPh fPcNGa regen-avatar"
      color="white"
      data-testid="avatar"
      role="avatar"
    >
      <span
        class="regen-avatar__text"
        style="transform: translateX(-50%) translateY(-50%) scale(1);"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar > should render rounded 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-beySPh jrJrEJ regen-avatar regen-avatar--rounded"
        color="white"
        data-testid="avatar"
        role="avatar"
      >
        <span
          class="regen-avatar__text"
          style="transform: translateX(-50%) translateY(-50%) scale(1);"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-beySPh jrJrEJ regen-avatar regen-avatar--rounded"
      color="white"
      data-testid="avatar"
      role="avatar"
    >
      <span
        class="regen-avatar__text"
        style="transform: translateX(-50%) translateY(-50%) scale(1);"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar > should render size 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-beySPh fPcNGa regen-avatar"
        color="white"
        data-testid="avatar"
        role="avatar"
      >
        <span
          class="regen-avatar__text"
          style="transform: translateX(-50%) translateY(-50%) scale(1);"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-beySPh fPcNGa regen-avatar"
      color="white"
      data-testid="avatar"
      role="avatar"
    >
      <span
        class="regen-avatar__text"
        style="transform: translateX(-50%) translateY(-50%) scale(1);"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar > should render src 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-beySPh fPcNGa regen-avatar"
        color="white"
        data-testid="avatar"
        role="avatar"
      >
        <img
          alt="avatar"
          loading="eager"
          src="https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50"
        />
        <span
          class="regen-avatar__text"
          style="transform: translateX(-50%) translateY(-50%) scale(1);"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-beySPh fPcNGa regen-avatar"
      color="white"
      data-testid="avatar"
      role="avatar"
    >
      <img
        alt="avatar"
        loading="eager"
        src="https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50"
      />
      <span
        class="regen-avatar__text"
        style="transform: translateX(-50%) translateY(-50%) scale(1);"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar > should render style 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-beySPh fPcNGa regen-avatar"
        color="white"
        data-testid="avatar"
        role="avatar"
        style="color: red;"
      >
        <span
          class="regen-avatar__text"
          style="transform: translateX(-50%) translateY(-50%) scale(1);"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-beySPh fPcNGa regen-avatar"
      color="white"
      data-testid="avatar"
      role="avatar"
      style="color: red;"
    >
      <span
        class="regen-avatar__text"
        style="transform: translateX(-50%) translateY(-50%) scale(1);"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
